src/include/common/types/types.h

```diff
@@ -274,6 +277,11 @@ class LogicalType {
     KUZU_API static std::vector<LogicalType> copy(const std::vector<LogicalType*>& types);
 
     static LogicalType ANY() { return LogicalType(LogicalTypeID::ANY); }
+    static LogicalType ANY(PhysicalTypeID physicalType) {
+        auto ret = LogicalType(LogicalTypeID::ANY);
+        ret.physicalType = physicalType;
+        return ret;
+    }
```

Comment:
I added this constructor to treat `ANY` as an internal logical type that contains a physical type of our choice. This is because logical types are needed to construct `Column` and `ColumnChunkData`.